src/litserve/server.py

```diff
@@ -80,10 +80,11 @@ def collate_requests(
 
         try:
             uid, timestamp, x_enc = request_queue.get(timeout=min(remaining_time, 0.001))
-            if time.monotonic() - timestamp <= lit_api.request_timeout:
+            if lit_api.request_timeout in [-1, False] or time.monotonic() - timestamp <= lit_api.request_timeout:
```

Comment:
This is quite wasteful done everytime, when we know everything ahead of time.

Just initialize a boolean `apply_timeout = lit_api.request_timeout not in [-1, False]` before the `while` loop so we can check `if apply_timeout and time.monotonic() - timestamp <= lit_api.request_timeout`

Also, we are calling `.append((uid, timestamp, x_enc))` but `timestamp` is not used anymore further down.
Let's avoid even including it in the tuple altogether.